packages/image-block/src/components/Image.tsx

```diff
@@ -49,23 +54,33 @@ export const ImageComponent = ({ image, blockSettings, isEditing, appBridge }: I
         className: joinClassNames(['tw-rounded', isFocused && FOCUS_STYLE]),
     };
 
-    return isEditing ? (
-        Image
-    ) : // eslint-disable-next-line unicorn/no-nested-ternary
-    link ? (
-        <a
-            {...props}
-            href={link.link.link}
-            target={link.openInNewTab ? '_blank' : undefined}
-            rel={link.openInNewTab ? 'noopener noreferrer' : 'noreferrer'}
-        >
-            {Image}
-        </a>
-    ) : (
-        <button {...props} onClick={() => open(image)}>
-            {Image}
-        </button>
-    );
+    const getImageComponent = () => {
+        if (isEditing) {
+            return Image;
+        }
+        if (link) {
+            return (
+                <a
+                    {...props}
+                    href={link.link.link}
+                    target={link.openInNewTab ? '_blank' : undefined}
+                    rel={link.openInNewTab ? 'noopener noreferrer' : 'noreferrer'}
+                >
+                    {Image}
+                </a>
+            );
+        }
+        if (isAssetViewerEnabled) {
+            return (
+                <button data-test-id="image-block-asset-viewer-button" {...props} onClick={() => open(image)}>
+                    {Image}
+                </button>
+            );
+        }
+        return Image;
+    };
+
+    return getImageComponent();
```

Comment:
```suggestion
        if (isEditing) {
        return Image;
    }

    if (link) {
        return (
            <a
                {...props}
                href={link.link.link}
                target={link.openInNewTab ? '_blank' : undefined}
                rel={link.openInNewTab ? 'noopener noreferrer' : 'noreferrer'}
            >
                {Image}
            </a>
        );
    }

    if (isAssetViewerEnabled) {
        return (
            <button data-test-id="image-block-asset-viewer-button" {...props} onClick={() => open(image)}>
                {Image}
            </button>
        );
    }

    return Image;
```

Still not good, but better

Reply:
Nice, adjusted